### ISSUE TITLE
[Table] visibleCount of virtual table is not reactive

### ISSUE THREAD

**Issue by litt1e-p** [Tencent/tdesign-vue-next#1374]
### tdesign-vue-next 版本

0.19.1

### 重现链接

https://stackblitz.com/edit/angular-kzfk22-um1jac?file=.stackblitzrc

### 重现步骤

1. switch to ***demo*** page by click button
2. scroll to the bottom by mouse dragging the scrollbar
3. click ***higher height*** button and mouse drag the scrollbar near to the bottom. the extra spaces will come to the bottom of the table finally
4. ![WeChatWork16600991233858.png](http://tva1.sinaimg.cn/large/005wcR5Jgy1h51hbhutbij315v0p57jj.jpg)

### 期望结果

no extra space exists

### 实际结果

it looks like `visibleCount` in `src/hooks/useVirtualScroll.ts` does not update as container offsetHeight updated, `refreshContainer()` should be called by `refreshTable()` after dom updated or other macro/micro task to be called. 
`refreshContainer` only calls once in `onMounted` may be the reason I guess.

### 补充说明

```ts
// src/table/base-table.tsx
// src/hooks/useVirtualScroll.ts
```

**Comment by litt1e-p** [comment]
env info:
```
"vue": "^3.2.37"
"tdesign-vue-next": "^0.19.1"
```

**Comment by chaishi** [comment]
copy that

**Comment by litt1e-p** [comment]
no fixed yet

**Comment by chaishi** [comment]
当前虚拟滚动存在一些问题，由于原开发者已放弃维护，现换其他人维护。虚拟滚动要求如下，

1. 支持不同高度
2. 支持滚动到特定元素，方便 Select 等组件定位到选中元素
3. 支持数据变化不重置，方便支持树形结构虚拟滚动

**Comment by litt1e-p** [comment]
thanks

**Comment by chaishi** [comment]
resolved in https://github.com/Tencent/tdesign-vue-next/pull/2112

待上述 PR 处理完成合并发布后（预计本周四发布），代码稍作修改，如下所示，

```
<div class="item">
      <t-table
        row-key="id"
        :columns="columns"
        :data="data"
        :bordered="true"
        :resizable="true"
        :stripe="true"
        :hover="true"
        :style="{ height: dynamicHeight }"
        height="100%"
        :is-fixed-row-height="true"
        :scroll="{ type: 'virtual', rowHeight: 30, bufferSize: 10 }"
      >
      </t-table>
    </div>
```

- `:style="{ height: dynamicHeight }"` 表示设置表格组件最外层元素高度为动态高度，
- `height="100%"` 表示设置表格内容高度随父元素

**Comment by chaishi** [comment]
published at `v0.26.2`
